File: src/lib/src/main/java/work/lclpnet/ap2/impl/util/scoreboard/TranslatedNumberFormat.java
package work.lclpnet.ap2.impl.util.scoreboard;

import net.minecraft.network.chat.numbers.NumberFormat;

/**
 * A {@link NumberFormat} that may differ per language, resolved for a viewer's language.
 */
@FunctionalInterface
public interface TranslatedNumberFormat {

    NumberFormat translateTo(String language);

    static TranslatedNumberFormat constant(NumberFormat format) {
        return language -> format;
    }
}

File: src/lib/src/main/kotlin/work/lclpnet/ap2/util/scoreboard/TranslatedNumberFormat.kt
package work.lclpnet.ap2.util.scoreboard

import net.minecraft.network.chat.numbers.NumberFormat

/**
 * A [NumberFormat] that may differ per language, resolved for a viewer's language.
 */
fun interface TranslatedNumberFormat {

    fun translateTo(language: String): NumberFormat?

    companion object {
        fun constant(format: NumberFormat?): TranslatedNumberFormat {
            return TranslatedNumberFormat { language -> format }
        }
    }
}
